src/main/java/core/basesyntax/RectangleSupplier.java

```diff
@@ -0,0 +1,14 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class RectangleSupplier {
+    private Random random = new Random();
+    private RandomSize size = new RandomSize();
+    private ColorSupplier color = new ColorSupplier();
+
+    public Figure getRandomRectangle() {
+        return new Rectangle(size.getRandomSize(),

```

Comment:
Creating a figure, don't pass expressions in the constructor. Instead, create separate variables for the width, length, and color, and then pass them to the constructor for better code readability.